ui/src/shared/reducers/notifications.js

```diff
@@ -1,47 +1,21 @@
-import u from 'updeep'
-import _ from 'lodash'
+import uuid from 'uuid'
+export const initialState = []
 
-export const notifications = (state = {}, action) => {
+export const notifications = (state = initialState, action) => {
   switch (action.type) {
-    case 'NOTIFICATION_RECEIVED': {
-      const {type, message} = action.payload
-      return u.updateIn(type, message, state)
-    }
-    case 'NOTIFICATION_DISMISSED': {
-      const {type} = action.payload
-      return u(u.omit(type), state)
-    }
-    case 'ALL_NOTIFICATIONS_DISMISSED': {
-      // Reset to initial state
-      return {}
+    case 'PUBLISH_NOTIFICATION': {
+      const notification = {
+        ...action.payload.notification,
+        id: uuid.v4(),
+      }
+      const newNotification = [notification]
```

Comment:
you could combine this with L7 above by wrapping that with brackets and calling it `newNotification`, and then you could destructure `{notification}` from `action.payload`

Reply:
```
      const notification = [
        {
          ...action.payload.notification,
          id: uuid.v4(),
        },
      ]

      return [...notification, ...state]
```

Reply:
yeah exactly. you coul call it `newNotification` and destructure `const {notification} = action.payload` as well, if you'd like. either way